### ISSUE TITLE
Evaluation failed: TypeError: Cannot read properties of undefined (reading 'LegacyPhoneFeatures')

### ISSUE THREAD

**Issue by hcalldee** [pedroslopez/whatsapp-web.js#1920]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'LegacyPhoneFeatures')
    at __puppeteer_evaluation_script__:13:76
    at ExecutionContext._evaluateInternal (C:\Users\ADMINAKRE\Documents\bahan development\coba_\node_modules\puppeteer\lib\cjs\puppeteer\common\ExecutionContext.js:221:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async ExecutionContext.evaluate (C:\Users\ADMINAKRE\Documents\bahan development\coba_\node_modules\puppeteer\lib\cjs\puppeteer\common\ExecutionContext.js:110:16)
    at async Client.initialize (C:\Users\ADMINAKRE\Documents\bahan development\coba_\node_modules\whatsapp-web.js\src\Client.js:267:9)

### Expected behavior

i want start with local authentication

### Steps to Reproduce the Bug or Issue

bug started when i implement LocalAuth Strategy when i call 
client on authenticated

### Relevant Code

const qrcode = require('qrcode-terminal');
const fs = require("fs")
const { Client, LegacySessionAuth, LocalAuth, MessageMedia} = require('whatsapp-web.js');
const client = new Client({
     authStrategy: new LocalAuth({
          clientId: "client-one" //Un identificador(Sugiero que no lo modifiques)
     })
})

// Save session values to the file upon successful auth
client.on('authenticated', (session) => {
    console.log(session);
});
 

### Browser Type

Google Chrome

### WhatsApp Account Type

Standard

### Does your WhatsApp account have multidevice enabled?

No, I am not using Multi Device

### Environment

OS: Windows
Phone OS: Android
whatsapp-web.js version 1.18.4
Node.js Version v18.12.1

### Additional context

i suspected, there is changed key on 
"window.Store.Features = window.mR.findModule('FEATURE_CHANGE_EVENT')[0].LegacyPhoneFeatures;"

**Comment by imorandin** [comment]
Same here

**Comment by valentinoConti** [comment]
Same here 😲, what could have changed that broke this?

**Comment by santiramirezc** [comment]
Me too :(

**Comment by elasti-co** [comment]
Same (

**Comment by coffemixer** [comment]
same problem, i'm waiting for solution here.
thanks :D

**Comment by ederyd79** [comment]
Sabe problem here!

**Comment by balinux** [comment]
my solution is
- delete .wwebjs_auth folder
- delete node_modules folder
- update package to version "whatsapp-web.js": "^1.18.4"
- run yarn install or npm install
- run app

- capture
<img width="507" alt="image" src="https://user-images.githubusercontent.com/10473273/211976158-42ec40f0-0f1a-4d3e-811d-fd129096bde9.png">

**Comment by coffemixer** [comment]
> my solution is
> 
> * delete .wwebjs_auth folder
> * delete node_modules folder
> * update package to version "whatsapp-web.js": "^1.18.4"
> * run yarn install or npm install
> * run app
> 
> nb : im using node version v16.18.1
> 
> * capture
> 
> <img alt="image" width="507" src="https://user-images.githubusercontent.com/10473273/211976158-42ec40f0-0f1a-4d3e-811d-fd129096bde9.png">

Thank you, problem solved :)

**Comment by hcalldee** [comment]
> my solution is
> 
> * delete .wwebjs_auth folder
> * delete node_modules folder
> * update package to version "whatsapp-web.js": "^1.18.4"
> * run yarn install or npm install
> * run app
> 
> nb : im using node version v16.18.1
> 
> * capture
> 
> <img alt="image" width="507" src="https://user-images.githubusercontent.com/10473273/211976158-42ec40f0-0f1a-4d3e-811d-fd129096bde9.png">

![image](https://user-images.githubusercontent.com/49970186/211983218-3ad5a997-3230-43a1-9956-7daa46c0e7b8.png)
Thks Worked !

**Comment by hcalldee** [comment]
> my solution is
> 
> * delete .wwebjs_auth folder
> * delete node_modules folder
> * update package to version "whatsapp-web.js": "^1.18.4"
> * run yarn install or npm install
> * run app
> 
> nb : im using node version v16.18.1
> 
> * capture
> 
> <img alt="image" width="507" src="https://user-images.githubusercontent.com/10473273/211976158-42ec40f0-0f1a-4d3e-811d-fd129096bde9.png">

only worked on first time, still got same error on next "node app" execution 
![image](https://user-images.githubusercontent.com/49970186/211984851-7c22c767-48d3-482b-b2f2-711c48f71ac7.png)

**Comment by ederyd79** [comment]
Same problem here

Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'LegacyPhoneFeatures')

**Comment by ederyd79** [comment]
This worked for me: 

https://github.com/pedroslopez/whatsapp-web.js/issues/1913#issuecomment-1379831986

**Comment by Tibinsunny** [comment]
Same issue. Downgrading might solve the problem

**Comment by hcalldee** [comment]
> Same issue. Downgrading might solve the problem

thats wont be a solution, i've tried before

**Comment by harsh0508** [comment]
Getting same issue . Sometimes it reads status and that is all.

**Comment by ayavuzerdogan** [comment]
> Bu benim için çalıştı:
> 
> [#1913 (yorum)](https://github.com/pedroslopez/whatsapp-web.js/issues/1913#issuecomment-1379831986)

i agree this solved the problem. also i am using centos 8

**Comment by Julzk** [comment]
this solved the problem:
https://github.com/Julzk/whatsapp-web.js/commit/668be3bd8442235530ee11f1cb397e929e2f670b

**Comment by brunomoreno7538** [comment]
> this solved the problem: [Julzk@668be3b](https://github.com/Julzk/whatsapp-web.js/commit/668be3bd8442235530ee11f1cb397e929e2f670b)

Worked for me, thanks!

**Comment by harsh0508** [comment]
> this solved the problem: [Julzk@668be3b](https://github.com/Julzk/whatsapp-web.js/commit/668be3bd8442235530ee11f1cb397e929e2f670b)

This works !

**Comment by iamsouravganguli** [comment]
Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'LegacyPhoneFeatures')
    at __puppeteer_evaluation_script__:13:76

**Comment by samuellea** [comment]
Getting same error as @iamsouravganguli

**Comment by arifintajul4** [comment]
> Getting same error as @iamsouravganguli - where can the **.wwebjs_auth** folder be found?

project root directori

**Comment by tasaddaq-rana** [comment]
same problem

**Comment by diogenesderm** [comment]
Esse ajuste resolvel meu problema. Na pasta node_modules/whatsapp-web.js/src/util/Injected.js: [Julzk@668be3b](https://github.com/Julzk/whatsapp-web.js/commit/668be3bd8442235530ee11f1cb397e929e2f670b)

**Comment by msolis6686** [comment]
SOTULION
https://github.com/pedroslopez/whatsapp-web.js/pull/1917/files/ccf9438a0c46f0387ad2c65604314f53b0b07e84#diff-de25cb4b9105890088bb033eac000d1dd2104d3498a8523082dc7eaf319738b8

**Comment by diogenesderm** [comment]
Antes: window.Store.Features = window.mR.findModule('FEATURE_CHANGE_EVENT')[0].LegacyPhoneFeatures;
    
    correcao: window.Store.Features = window.mR.findModule('legacyPhoneFeatures')[0];

**Comment by renants286** [comment]
I'm using whatsapp-web.js on node-red...

After fixing #1913 I am getting a new error:
https://github.com/raweee/node-red-contrib-whatsapp-link/issues/8

 Someone can help me?

**Comment by Lucas-Angelo** [comment]
> * node_modules

Thank you, its work!

**Comment by Lucas-Angelo** [comment]
> my solution is
> 
> * delete .wwebjs_auth folder
> * delete node_modules folder
> * update package to version "whatsapp-web.js": "^1.18.4"
> * run yarn install or npm install
> * run app
> 
> nb : im using node version v16.18.1
> 
> * capture
> 
> <img alt="image" width="507" src="https://user-images.githubusercontent.com/10473273/211976158-42ec40f0-0f1a-4d3e-811d-fd129096bde9.png">

Thank u, its work!

**Comment by ederyd79** [comment]
Editing this: window.Store.Features = window.mR.findModule('legacyPhoneFeatures')[0]; just works for individual chats, but not for group chats: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'sendAddParticipants')

Any way to resolve the problem of managing Groups?

**Comment by T3uZ** [comment]
> Editando isto: window.Store.Features = window.mR.findModule('legacyPhoneFeatures')[0]; funciona apenas para bate-papos individuais, mas não para bate-papos em grupo: Falha na avaliação: TypeError: Não é possível ler as propriedades de indefinido (lendo 'sendAddParticipants')
> 
> Alguma maneira de resolver o problema de gerenciamento de grupos?

Deixa comentado essa linha, e tenta mandar mensagem no grupo, manda o feedback aqui.

**Comment by otomay** [comment]
> my solution is
> 
> * delete .wwebjs_auth folder
> * delete node_modules folder
> * update package to version "whatsapp-web.js": "^1.18.4"
> * run yarn install or npm install
> * run app
> 
> nb : im using node version v16.18.1
> 
> * capture
> 
> <img alt="image" width="507" src="https://user-images.githubusercontent.com/10473273/211976158-42ec40f0-0f1a-4d3e-811d-fd129096bde9.png">

funcionando

**Comment by BETOXL** [comment]
Tengo mantado en un node 14.20.1 y tengo el mismo problema. Voy a esperar una nueva versión donde se solucione esto.

**Comment by slottler** [comment]
I am suffering from the same problem.  My code worked a few days back, but now it doesn't.  I'm super-grateful for whatsapp-web.js ... thank you so much!  But I really don't want to start editing libraries.  I humbly and hopefully ask that this can be fixed?
- Thank you very kindly!  :-)

**Comment by mdtausifiqbal** [comment]
### Same Issue with me

```
H:\Projects\WhatsappWeb\node_modules\puppeteer\lib\cjs\puppeteer\common\ExecutionContext.js:221
            throw new Error('Evaluation failed: ' + helper_js_1.helper.getExceptionMessage(exceptionDetails));
                  ^

Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'LegacyPhoneFeatures')
    at __puppeteer_evaluation_script__:13:76
    at ExecutionContext._evaluateInternal (H:\Projects\WhatsappWeb\node_modules\puppeteer\lib\cjs\puppeteer\common\ExecutionContext.js:221:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (H:\Projects\WhatsappWeb\node_modules\puppeteer\lib\cjs\puppeteer\common\ExecutionContext.js:110:16)
    at async Client.initialize (H:\Projects\WhatsappWeb\node_modules\whatsapp-web.js\src\Client.js:267:9)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

H:\Projects\WhatsappWeb>
```

Kindly fix it, it is showing this error after scanning QRCode, I am using updated Whatsapp and whatsapp-web.js

**Comment by sachhu** [comment]
I am also facing the same issue.
I tried deleting node_modules and .wwebjs_auth and running npm install.

I got below warning while running npm install 
`
npm WARN deprecated puppeteer@13.7.0: < 18.1.0 is no longer supported

added 151 packages, and audited 152 packages in 44s

22 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
`

Maybe puppeteer is outdated because of this we are facing this issue.

Please try to fix this issue as soon as possible.

**Comment by ederyd79** [comment]
> > Editando isto: window.Store.Features = window.mR.findModule('legacyPhoneFeatures')[0]; funciona apenas para bate-papos individuais, mas não para bate-papos em grupo: Falha na avaliação: TypeError: Não é possível ler as propriedades de indefinido (lendo 'sendAddParticipants')
> > Alguma maneira de resolver o problema de gerenciamento de grupos?
> 
> Deixa comentado essa linha, e tenta mandar mensagem no grupo, manda o feedback aqui.
> 
> Pode tentar fazer isso tambem: window.Store.Features = window.mR.findModule('FEATURE_CHANGE_EVENT')[0]?.LegacyPhoneFeatures;

Thanks for the replay but the problem is not sending messages, but creating, and editing groups, and adding or removing participants from them.

**Comment by hamidsaifi77** [comment]
Hi i got the solution of this by doing:-
This
window.Store.Features = window.mR.findModule('FEATURE_CHANGE_EVENT')[0].LegacyPhoneFeatures;

to

This
window.Store.Features = window.mR.findModule('FEATURE_CHANGE_EVENT')[0]?.LegacyPhoneFeatures;

**Comment by sachhu** [comment]
Pull request #1917 resolves the issue

**Comment by ruanduarte** [comment]
this works for me too.. o/

**Comment by augustolok** [comment]
tenta usar isso 
 try {        
        window.Store.Features = window.mR.findModule('FEATURE_CHANGE_EVENT')[0].LegacyPhoneFeatures;
    } catch (error) {
        
    }

**Comment by TakerOver** [comment]
after deleting the .wwebjs_auth/ folder it worked for 6-12 hours i repeated the procedure 8 times now even if i delete .wwebjs_auth/ and scan qrcode immediately 'LegacyPhoneFeatures' error.
At the moment whatsapp-web.js doesn't work anymore.

**Comment by TakerOver** [comment]
issues/1920 it works

**Comment by TakerOver** [comment]
> Hi, i got the solution to this by doing:-
> 
> Path:- node_modules/whatsapp-web.js/src/util/Injected.js Line number 16
> 
> This window.Store.Features = window.mR.findModule('FEATURE_CHANGE_EVENT')[0].LegacyPhoneFeatures;
> 
> to
> 
> This window.Store.Features = window.mR.findModule('FEATURE_CHANGE_EVENT')[0]?.LegacyPhoneFeatures;

issues/1920 it works

**Comment by anthonywebster** [comment]
same problem

**Comment by sharon846** [comment]
Still, nothing works. I have to work with groups creation and there is no solution..

**Comment by anthonywebster** [comment]
You can use the solution here: #1917 , it basically consists of using Optional chaining, then you can go to the file and make the change yourself, normally the node_module folder is ignored so the changes won't go away and I use a package called patch-package to be able to publish the changes while the pull request is being merged.

![image](https://user-images.githubusercontent.com/3496319/212546647-c5a6b6de-6c67-4b3e-b431-cae849b3d346.png)

Here I leave a small tutorial that I found on how to use the patch-package package, I have my code hosted on heroku.

https://dev.to/zhnedyalkow/the-easiest-way-to-patch-your-npm-package-4ece

**Comment by sharon846** [comment]
Thanks! You are correct that it removes that error of Legacy, but it makes other errors, because it fanboy find the creatGroup function etc.. in other words - it doesn’t solve my problem (I’ve tried everything they suggested)

**Comment by RickyFirmansyah27** [comment]
npm WARN deprecated puppeteer@13.7.0: < 18.1.0 is no longer supported after delete node_modules and go install back

**Comment by RickyFirmansyah27** [comment]
> window.Store.Features = window.mR.findModule('FEATURE_CHANGE_EVENT')[0]?.LegacyPhoneFeatures;

thanks bruh its work

**Comment by ratxrat** [comment]
im just update wa-web-js to latest version and solve error

**Comment by pedroslopez** [comment]
This was a duplicate of #1913 

A fix has now been released! Please update to v1.19.1 to address the issue.

**Comment by paitoncamp** [comment]
> this solved the problem: [Julzk@668be3b](https://github.com/Julzk/whatsapp-web.js/commit/668be3bd8442235530ee11f1cb397e929e2f670b)

this work for me

**Comment by iamsouravganguli** [comment]
> 
yes
